kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/SolverServicesClient.java

```diff
@@ -41,6 +42,17 @@ void solvePlanningProblem(String containerId,
     void terminateSolverEarly(String containerId,
                               String solverId);
 
+    void addProblemFactChange(String containerId,
+                              String solverId,
+                              ProblemFactChange problemFactChange);
+
+    void addProblemFactChanges(String containerId,
+                              String solverId,
+                              List<ProblemFactChange> problemFactChange);
+
+    Boolean isEveryProblemFactChangeProcessed(String containerId,
+                                              String solverId);
```

Comment:
Ok, so we can't return boolean, we need to return Boolean due to jms limitations?
I can live with that.